### PR TITLE
Ready to monitor EC2?

### PR DESCRIPTION
**Pull request by mattdodge**

### CODE REVIEW

**Comment by mattdodge** [comment]
Hang on, making a change to this that allows us to capture all AWS metrics, not just EC2

**Comment by mattdodge** [comment]
Ok, added. Back to "ready for review" state

File: cloud_watch_block.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        namespace, dimensions, and possibly some other important information
+        for querying later.
+
+        Returns:
+            None: saves the metics in the self._metrics list instead
+        """
+        with self._metrics_lock:
+            self._logger.debug(
+                "Syncing valid metrics for {} from EC2".format(self.metric))
+            metrics_res = self._conn.list_metrics(metric_name=self.metric)
```
Comment (hansmosh) [inline cloud_watch_block.py:94]: If credentials are invalid:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/boto/ec2/cloudwatch/**init**.py", line 286, in list_metrics
    return self.get_list('ListMetrics', params, [('member', Metric)])
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/boto/connection.py", line 1182, in get_list
    raise self.ResponseError(response.status, response.reason, body)
boto.exception.BotoServerError: BotoServerError: 403 Forbidden
<ErrorResponse xmlns="http://monitoring.amazonaws.com/doc/2010-08-01/">
  <Error>
    <Type>Sender</Type>
    <Code>InvalidClientTokenId</Code>
    <Message>The security token included in the request is invalid.</Message>
  </Error>
  <RequestId>37425fea-e208-11e4-ac8a-9dde9c0aa0ed</RequestId>
</ErrorResponse>

**Comment by hansmosh** [comment]
Looks like the block just needs one little update. If you are using invalid credentials, then list_metrics raises an exception. It is probably valid that it raises the exception in configure because that will properly prevent the service from starting, but it should be caught and logged and then re-raised.

```
conn = connect_to_region('us-east-1', aws_access_key_id='asdf')
conn.list_metrics(metric_name='CPUCreditBalance')
```

**Pull request merged**